gbpservice/nfp/config_orchestrator/agent/loadbalancerv2.py

```diff
@@ -0,0 +1,457 @@
+                              resource_id, lb_id,
+                              service_type):
+        request_data = None
+        try:
+            request_data = common.get_network_function_map(
+                context, nf_id)
+            # Adding Service Type #
+            # TODO(jiahao): is the key name "loadbalancer_id" correct?
+            request_data.update({"service_type": service_type,
+                                 "loadbalancer_id": lb_id,
+                                 "neutron_resource_id": resource_id,
+                                 "LogMetaID": nf_id})
+        except Exception as e:
+            LOG(LOGGER, 'ERROR', '%s' % (e))
+            return request_data
+        return request_data
+
+    # TODO(jiahao): copy from v1 agent need to review
+    def _trigger_service_event(self, context, event_type, event_id,
+                               request_data):
+        event_data = {'resource': None,
+                      'context': context.to_dict()}
+        event_data['resource'] = {'eventtype': event_type,
+                                  'eventid': event_id,
+                                  'eventdata': request_data}
+        ev = self._sc.new_event(id=event_id,
+                                key=event_id, data=event_data)
+        self._sc.post_event(ev)
+
+    def update_status(self, context, notification_data):
+        notification = notification_data['notification'][0]
+        notification_info = notification_data['info']
+        resource_data = notification['data']
+        obj_type = resource_data['obj_type']
+        obj_id = resource_data['obj_id']
+        service_type = notification_info['service_type']
+
+        rpcClient = transport.RPCClient(a_topics.LBV2_NFP_PLUGIN_TOPIC)
+        rpcClient.cctxt = rpcClient.client.prepare(
+            version=const.LOADBALANCERV2_RPC_API_VERSION)
+
+        lb_p_status = const.ACTIVE
+        lb_o_status = None
+        obj_p_status = resource_data['provisioning_status']
+        obj_o_status = resource_data['operating_status']
+
+        msg = ("NCO received LB's update_status API, making an update_status "
+               "RPC call to plugin for %s: %s with status %s" % (
+                   obj_type, obj_id, obj_p_status))
+        LOG(LOGGER, 'INFO', '%s' % (msg))
+
+        if obj_type == 'healthmonitor':
+                obj_o_status = None
+
+        if obj_type != 'loadbalancer':
+            rpcClient.cctxt.cast(context, 'update_status',
+                                 obj_type=obj_type,
+                                 obj_id=obj_id,
+                                 provisioning_status=obj_p_status,
+                                 operating_status=obj_o_status)
+        else:
+            lb_o_status = const.ONLINE
+            if obj_p_status == const.ERROR:
+                lb_p_status = const.ERROR
+                lb_o_status = const.OFFLINE
+
+        rpcClient.cctxt.cast(context, 'update_status',
```

Comment:
Do we need to update loadbalancer resource status all the time even if resource is not loadbalancer ?